src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1391,6 +1697,218 @@ public override DataTable GetSchemaTable()
 
         #endregion Schema metadata table
 
+        #region Seeking
+
+        Task SeekToColumn(int column, bool async)
+        {
+            if (_isSequential)
+                return SeekToColumnSequential(column, async);
+            else
+            {
+                SeekToColumnNonSequential(column);
+                return PGUtil.CompletedTask;
+            }
+        }
+
+        void SeekToColumnNonSequential(int column)
+        {
+            if (_column == -1)
+            {
+                // Starting a new row.
+
+                // We currently assume that the row's number of columns is identical to the description's
+#if DEBUG
+                _numColumns = Buffer.ReadInt16();
+                Debug.Assert(RowDescription.NumFields == _numColumns);
+#else
+                Buffer.ReadPosition += 2;
+#endif
+
+                // Initialize our columns array with the offset and length of the first column
+                _columns.Clear();
+                var len = Buffer.ReadInt32();
+                _columns.Add((Buffer.ReadPosition, len));
+            }
+
+            // Shut down any streaming going on on the column
+            if (ColumnStream != null)
+            {
+                ColumnStream.Dispose();
+                ColumnStream = null;
+                // Disposing the stream leaves us at the end of the column
+                PosInColumn = ColumnLen;
```

Comment:
You don't need to set `PosInColumn` here since it's reset to zero at the end of the method.

Reply:
Removed.